src/matrix/room/Room.js

```diff
@@ -250,6 +259,32 @@ export class Room extends BaseRoom {
         }
     }
 
+    _updateObservedMembers(memberChanges) {
+        for (const [userId, memberChange] of memberChanges) {
+            const observableMember = this._observedMembers.get(userId);
+            if (observableMember) {
+                observableMember.set(memberChange.member);
+            }
+        }
+    }
+
+    _getPowerLevelsEvent(roomResponse) {
+        const isPowerlevelEvent = event => event.type === POWERLEVELS_EVENT_TYPE;
```

Comment:
can we also check here there is a `state_key` on the event, to make sure people can't sneak non-state events in the power levels